src/DurableTask.Netherite/StorageLayer/Faster/FasterKV.cs

```diff
@@ -186,7 +186,7 @@ void Shutdown()
         ClientSession<Key, Value, EffectTracker, Output, object, IFunctions<Key, Value, EffectTracker, Output, object>> CreateASession(string id, bool isScan)
         {
             var functions = new Functions(this.partition, this, this.cacheTracker, isScan);
-            return this.fht.NewSession(functions, id, readFlags: (isScan ? ReadFlags.None : ReadFlags.CopyReadsToTail));
```

Comment:
so how did this get dropped?

Reply:
`NewSession()`doesn't take an argument named `readFlags` after upgrading the FASTER version so it seems like there's an API change here. There's a `ReadCopyOptions` parameter though so maybe this information should be sent through that. This would be good to clarify with the FASTER team.

Reply:
Correct--we've improved the control over copy source/destination in later versions. Here you probably want:
   isScan ? default : new {CopyFrom = ReadCopyFrom.Device, CopyTo = ReadCopyFrom.MainLog};

Reply:
To keep it consistent with the current Netherite design I think we should copy from the immutable region also. So I would suggest something like this:

```csharp
            ReadCopyOptions readCopyOptions = isScan 
                ? new ReadCopyOptions(ReadCopyFrom.None, ReadCopyTo.None) 
                : new ReadCopyOptions(ReadCopyFrom.AllImmutable, ReadCopyTo.MainLog);
            
            return this.fht.NewSession(functions, id, default, readCopyOptions);
```